src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -385,6 +385,9 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                 {
                     // Run solution based up to date check.
                     var projectsNeedingRestore = _solutionUpToDateChecker.PerformUpToDateCheck(originalDgSpec, _logger).AsList();
+                    var specialReferencesCount = originalDgSpec.Projects
+                        .Where(x => x.RestoreMetadata.ProjectStyle != ProjectStyle.PackageReference && x.RestoreMetadata.ProjectStyle != ProjectStyle.PackagesConfig && x.RestoreMetadata.ProjectStyle != ProjectStyle.ProjectJson)
```

Comment:
Can we check if `x.RestoreMetadata.ProjectStyle` is a `special reference type` instead of multiple conidtions in the `where` clause?

Reply:
`.ProjectStyle` is an enum with multiple possible values. I'm trying to understand and see if I can use a switch case to improve code readability

Reply:
How would a switch improver readability in this case? 
You can just put each enum value check on a different line to improve readability, cause that's effectively what a switch would do, but you'd need a loop as well.

Reply:
my bad that I didn't check the conditions properly. sorry for the confusion. let's leave existing condition as is.